index.js

```diff
@@ -30,31 +32,52 @@ function consulClient(host) {
     var requestOptions = prepareRequestOptions(settings);
     var healthUrl = composeConsulHealthUrl(
       host,
-      settings.serviceName,
-      settings.version
+      settings.serviceName
     );
+    var version = semver.validRange(settings.version);
 
     // Make a request to Consul host to retrieve service health info
     // selects a healthy service, if one exists
     // uses service data found to compose a url of where to reach the service
     // makes an HTTP request to the service.
     return got(healthUrl, { json: true })
-      .then(selectServiceInstance)
+      .then(function(response) {
+        return selectServiceInstance(response, version)
+      })
       .then(serviceUrlComposer(settings.endpoint))
       .then(makeRequest);
 
     /**
      * Selects a healthy service instance to use if one exists.
      *
      * @param {object} response
+     * @param {String} version
      * @returns {Service|*}
      */
-    function selectServiceInstance(response) {
+    function selectServiceInstance(response, version) {
       if (isEmpty(response.body)) {
-        throw new Error('no service instances available');
+        throw new Error('No service instances available');
       }
 
-      return sample(response.body).Service;
+      if (!version) {
+        throw new Error('Invalid version supplied');
+      }
+
+      var matches = [];
+
+      response.body.forEach(function(hit) {
+        if (semver.satisfies(hit.Service.Tags[0] = hit.Service.Tags[0].replace(/-/g, '.'), version)) {
```

Comment:
The inline assignment here is a little odd -- I think it is intentional so that the filter function on line 80 matches correctly, but I think I would pull the assignment out of the `if` / `satisfies` call and just do it on it's own before the `if`. It's really easy to miss that `=` lurking in the middle of all the other functionality on this line.